File: db4oj.tests/src/com/db4o/db4ounit/common/io/ReadOnlyBinTest.java
package com.db4o.db4ounit.common.io;

import com.db4o.ext.*;
import com.db4o.io.*;

import db4ounit.*;

public class ReadOnlyBinTest extends StorageTestUnitBase {
	
	public void test() {
		reopenAsReadOnly();
		assertReadOnly(_bin);
	}

	private void reopenAsReadOnly() {
	    close();
		open(true);
    }
	
	private void assertReadOnly(final Bin adapter) {
		Assert.expect(Db4oIOException.class, new CodeBlock() {
			public void run() throws Throwable {
				adapter.write(0, new byte[] {0}, 1);
			}
		});
	}
}

File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/IO/ReadOnlyBinTest.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4objects.Db4o.Ext;
using Db4objects.Db4o.IO;
using Db4objects.Db4o.Tests.Common.IO;

namespace Db4objects.Db4o.Tests.Common.IO
{
	public class ReadOnlyBinTest : StorageTestUnitBase
	{
		public virtual void Test()
		{
			ReopenAsReadOnly();
			AssertReadOnly(_bin);
		}

		private void ReopenAsReadOnly()
		{
			Close();
			Open(true);
		}

		private void AssertReadOnly(IBin adapter)
		{
			Assert.Expect(typeof(Db4oIOException), new _ICodeBlock_21(adapter));
		}

		private sealed class _ICodeBlock_21 : ICodeBlock
		{
			public _ICodeBlock_21(IBin adapter)
			{
				this.adapter = adapter;
			}

			/// <exception cref="System.Exception"></exception>
			public void Run()
			{
				adapter.Write(0, new byte[] { 0 }, 1);
			}

			private readonly IBin adapter;
		}
	}
}
